examples/workflows/cockroach_basic_cluster/main.tf

```diff
@@ -0,0 +1,87 @@
+variable "cluster_name" {
+  type     = string
+  nullable = false
+}
+
+variable "sql_user_name" {
+  type     = string
+  nullable = false
+  default  = "maxroach"
+}
+
+# Remember that even variables marked sensitive will show up
+# in the Terraform state file. Always follow best practices
+# when managing sensitive info.
+# https://developer.hashicorp.com/terraform/tutorials/configuration-language/sensitive-variables#sensitive-values-in-state
+variable "sql_user_password" {
+  type      = string
+  nullable  = false
+  sensitive = true
+}
+
+variable "cloud_provider" {
+  type     = string
+  nullable = false
+  default  = "GCP"
+}
+
+variable "cloud_provider_regions" {
+  type     = list(string)
+  nullable = false
+  default  = ["us-central1"]
+}
+
+terraform {
+  required_providers {
+    cockroach = {
+      source = "cockroachdb/cockroach"
+    }
+  }
+}
+provider "cockroach" {
+  # export COCKROACH_API_KEY with the cockroach cloud API Key
+}
+
+resource "cockroach_cluster" "example" {
+  name           = var.cluster_name
+  cloud_provider = var.cloud_provider
+  plan           = "BASIC"
+  serverless     = {}
```

Comment:
If the plan is BASIC can we imply the "serverless" field? I think maybe it would be an error for them to have anything within the brackets.

Reply:
They can have `usage_limits` in the brackets. If they have nothing, it indicates there are no limits enforced on their Basic Serverless cluster. It would be possible to imply the "serverless" field if the plan is BASIC, but that would be a different change. @marksoper for visibility.